### ISSUE TITLE
[ban-types] Don't ban Function unless the function is invoked

### ISSUE THREAD

**Issue by cowwoc** [typescript-eslint/typescript-eslint#2717]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

ban-types currently warns on the use of `Function` but the check is overly simple, resulting in false positives. Given:

```
class Checker {
  /**
   * @param first a value
   * @param second the type of a value
   * @return true if "first" is an instance of type "second"
   */
  public instanceOf(first: unknown, second: Function): boolean {
    return first instanceof second;
  }
}
```

The implementation never tries instantiating `second`, hence the rule should not complain. As far as I can tell, there is no way to define the function shape in this case.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.5.0` |
| `@typescript-eslint/parser`        | 4.5.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.11.0` |
| `node`                             | `14.11.0` |

**Comment by bradzacher** [comment]
The rule is intentionally simple and configurable.
It simply bans usage of a specified named type.

This is working as intended and the rule's functionality will not be special cased for specific named types.
